code/game/objects/items/grenades/atmos_grenades.dm

```diff
@@ -97,3 +97,40 @@
 		var/turf/open/floor_loc = turf_loc
 		floor_loc.atmos_spawn_air("n2o=[n2o_gas_amount / distance_from_center];TEMP=273")
 	qdel(src)
+
+/obj/item/grenade/gas_crystal/crystal_foam
+	name = "crystal foam"
+	desc = "A crystal with a foggy inside"
+	icon_state = "crystal_foam"
+	var/list/obj/item/reagent_containers/glass/beakers = list()
+	var/breach_range = 7
+
+/obj/item/grenade/gas_crystal/crystal_foam/Initialize()
+	. = ..()
+	var/obj/item/reagent_containers/glass/beaker/large/first_beaker = new(src)
+	var/obj/item/reagent_containers/glass/beaker/second_beaker = new(src)
+
+	first_beaker.reagents.add_reagent(/datum/reagent/aluminium, 75)
+	second_beaker.reagents.add_reagent(/datum/reagent/smart_foaming_agent, 25)
+	second_beaker.reagents.add_reagent(/datum/reagent/toxin/acid/fluacid, 25)
```

Comment:
why

Reply:
Also are you sure that these will properly delete when the crystal deletes?

Reply:
This is how the smart foam grenade does. I'll add something to Destroy() just to be sure

Reply:
That's because the smart foam grenade is a subclass of chemical grenades. This here isn't, so unless you're going to add some way to take the beakers out (why would you do that it's a crystal) you can just make it create the reagents upon detonation.

Reply:
Ahhh I see now, I'll do that

Reply:
you still plan on doing that?